### DISCUSSION TITLE
how to commit changes generated by a `version` hook?

### DISCUSSION THREAD

**Discussion by osaxma** [invertase/melos#361] [Q&A] [answered]
In my cli package, I've created an `update_version.dart` script that generates the new version number in a `version.g.dart` file within my cli package. The script is hooked to `version` (similar to how `melos` does it). 

After I run `melos version`, the `version.g.dart` is left ~in the staging area~ so it's not included with the release commit (hence, the release contains a wrong version). 

The only solution that I could think of is amending the changes to the release commit before publishing.

I was just wondering if there's a more streamlined way? or how does `melos` do it?

---
I would also like to take the chance and thank you all for the awesome package by the way -- huge time saver and works like a charm (never had such a clean changelog before 😅)

**Comment by blaugold** [comment, marked as answer]
Thanks @osaxma! Great to hear that Melos is helping you.

For the `melos` package, we currently have some [special handling](https://github.com/invertase/melos/blob/16fe691665466e81e34c87451e8b1ca32809bf95/packages/melos/lib/src/commands/version.dart#L776-L785).

In your `version` hook script, you should add the generated files to git's staging area after updating them. That way, changes to those files will be included in the commit created by Melos. That's how I do it other packages that I maintain with Melos.

**Reply by osaxma** [reply]
> For the melos package, we currently have some [special handling](https://github.com/invertase/melos/blob/16fe691665466e81e34c87451e8b1ca32809bf95/packages/melos/lib/src/commands/version.dart#L776-L785).

Now that makes sense!! 

Thanks a lot @blaugold  🙏🏽
